doc/topics/development/contributing.rst

```diff
@@ -4,481 +4,604 @@
+`keepachangelog <https://keepachangelog.com/en/1.0.0/>`__ approach for
+our changelog, and use towncrier to generate it for each release. As a
+contributor, all that means is that you need to add a file to the
+``salt/changelog`` directory, using the ``<issue #>.<type>`` format. For
+instanch, if you fixed issue 123, you would do:
 
-.. _backporting-pull-requests:
+::
 
+   echo "Made sys.doc inform when no minions return" > changelog/123.fixed
 
-Backporting Pull Requests
--------------------------
+And that’s all that would go into your file. When it comes to your
+commit message, it’s usually a good idea to add other information - what
+does a reviewer need to know about the change that you made? If someone
+isn’t an expert in this area, what will they need to know?
 
-On rare occasions, a serious bug will be found in the middle of a release
-cycle. These bugs will require a point release. Contributors should still
-submit fixes directly to ``master``, but they should also call attention to the
-fact that it addresses a critical issue and will need to be back-ported.
+This will also help you out, because when you go to create the PR it
+will automatically insert the body of your commit messages.
 
-Keeping Salt Forks in Sync
---------------------------
+PR Time!
+--------
 
-Salt advances quickly. It is therefore critical to pull upstream changes from
-upstream into your fork on a regular basis. Nothing is worse than putting hard
-work into a pull request only to see bunches of merge conflicts because it has
-diverged too far from upstream.
+Once you’ve done all your dev work and tested locally, you should check
+out our `PR
+guidelines <https://docs.saltstack.com/en/develop/topics/development/pull_requests.html>`__.
+After you read that page, it’s time to `open a new
+PR <https://github.com/saltstack/salt/compare>`__. Fill out the PR
+template - you should have updated or created any necessary docs, and
+written tests if you’re providing a code change. When you submit your
+PR, we have a suite of tests that will run across different platforms to
+help ensure that no known bugs were introduced.
 
-.. seealso:: `GitHub Fork a Repo Guide`_
+Now What?
+~~~~~~~~~
 
-The following assumes ``origin`` is the name of your fork and ``upstream`` is
-the name of the main `saltstack/salt`_ repository.
+You’ve made your changes, added documentation, opened your PR, and have
+passing tests… now what? When can you expect your code to be merged?
 
-#.  View existing remotes.
+When you open your PR, a reviewer will get automatically assigned. If
+your PR is submitted during the week you should be able to expect some
+kind of communication within that business day. If your tests are
+passing and we’re not in a code freeze, ideally your code will be merged
+that day. If you haven’t heard from your assigned reviewer, ping them on
+GitHub, `irc <https://webchat.freenode.net/#salt>`__, or Community
+Slack.
 
```

Comment:
The goal is that we should at least be able to leave some sort of feedback within a business day. That's definitely not been the case typically, but that's not because it can't be done - we just haven't prioritized it.

Reply:
I like your "can-do" attitude. Feel free to leave this as-is. :)